### ISSUE TITLE
class in GraphQL namespace causes null ref in AllowedOnAnalyzer

### ISSUE THREAD

**Issue by SimonCropp** [graphql-dotnet/graphql-dotnet#4031]
version 8

this class 

```
namespace GraphQL;

public static class MyClass
{
    public static string Method() =>
        nameof(MyClass.Method);
}
```

causes 

```
  Analyzer 'GraphQL.Analyzers.AllowedOnAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
```

repro https://github.com/SimonCropp/GraphQLNullRepo

**Comment by gao-artur** [comment]
This is actually related to the assembly name, not to the namespace. And it happens because `IsGraphQLSymbol` returns `true`
https://github.com/graphql-dotnet/graphql-dotnet/blob/610c375663fc3ffc61395a81dffc9a67a9e90408/src/GraphQL.Analyzers/Helpers/GraphQLExtensions.cs#L31-L33

Need a better way to decide if the symbol is defined by GraphQL.NET or not. I'd prefer not to enumerate all the possible assembly names explicitly because then I need to remember to extend this list every time a new project is added.
@Shane32 any ideas?

**Comment by Shane32** [comment]
With the Shouldly test package, sometimes I want to write another ShouldSomething extension. It’s nice that the code detects that it’s in a Shouldly namespace (even though not actually the same assembly) and then treats it like any other shouldly extension. 

So what I’m saying is that it may not be bad necessarily if it returns true in that case. Sorta depends on how it’s used I guess.

**Comment by Shane32** [comment]
> because then I need to remember to extend this list every time a new project is added.

in the problem today, the assembly is named exactly GraphQL.dll right ? So you couldn’t tell the difference from the assembly name alone?

**Comment by Shane32** [comment]
Why does the allowed on analyzer care what assembly? I’m not at a pc yet; maybe it’s obvious.

**Comment by gao-artur** [comment]
All analyzers start by finding a relevant method, class, etc., by its name. Then, they check that the symbol they find is defined by the GraphQL.NET library and not by the user code (think `FieldBuilder` class, but not related to GraphQL.NET). This is done to avoid reporting our diagnostics on the user types by mistake.
Today we have at least 2 assemblies relevant for analyzers: `GraphQL` and `GraphQL.MicrosoftDI` (`AllowedOnAttribute` is used in both these projects).

**Comment by gao-artur** [comment]
The actual bug was in the analyzer implementation, not in the `IsGraphQLSymbol` extension method. But it was discovered because `IsGraphQLSymbol` returned true. There is a high risk of false positives in such assemblies. Worse than that, the code fixes can break the user code if they think these symbols are from our library.

**Comment by Shane32** [comment]
Maybe we could add a GraphQL-specific attribute to our assemblies to assist with identification of the assembly as being from GraphQL.NET...?

for example:

```cs
[assembly: System.ComponentModel.Description("__GraphQL_NET_Assembly__")]
```

**Comment by gao-artur** [comment]
Sounds like a plan. I like it!

**Comment by Shane32** [comment]
I think there's a way to add an assembly-level attribute into the directory.build.targets file so it will apply to all IsPackable projects.  I'm not sure how but ChatGPT came up with this:

```xml
<Project>
  <Target Name="AddGlobalAssemblyInfo" BeforeTargets="BeforeBuild" Condition="'$(IsPackable)' == 'true'">
    <ItemGroup>
      <AssemblyAttribute Include="System.ComponentModel.Description">
        <_Parameter1>__GraphQL_NET_Assembly__</_Parameter1>
      </AssemblyAttribute>
    </ItemGroup>
  </Target>
</Project>
```

**Comment by Shane32** [comment]
you can of course alter it as desired, just an idea.

**Comment by SimonCropp** [comment]
do u use the same snk for all assemblies?

**Comment by Shane32** [comment]
I believe so

**Comment by SimonCropp** [comment]
so could the snk be checked to ensure is is a core graphql package?

**Comment by Shane32** [comment]
Probably. Can you read it via reflection? I’ve never tried to read a snk before.

**Comment by SimonCropp** [comment]
@Shane32 https://github.com/graphql-dotnet/graphql-dotnet/pull/4035

**Comment by Shane32** [comment]
Nice

**Comment by SimonCropp** [comment]
fixed in 8.0.1. thanks peoples
